from: https://github.com/hexdump0815/linux-mainline-and-mali-generic-stable-kernel/issues/6

From 170fa62ecc113a44cf33591c7b38643eaf457c55 Mon Sep 17 00:00:00 2001
From: Anton Bambura <jenneron@protonmail.com>
Date: Wed, 5 Jan 2022 16:13:53 +0200
Subject: [PATCH] ARM: dts: exynos: snow: add atmel mxt touchpad

Add Atmel mXT touchpad (like peach chromebooks) as it is used in
some revisions (probably rev5 only).

Signed-off-by: Anton Bambura <jenneron@protonmail.com>
---
 arch/arm/boot/dts/exynos5250-snow-common.dtsi | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/boot/dts/exynos5250-snow-common.dtsi b/arch/arm/boot/dts/exynos5250-snow-common.dtsi
index c952a6151..83c1a3a4e 100644
--- a/arch/arm/boot/dts/exynos5250-snow-common.dtsi
+++ b/arch/arm/boot/dts/exynos5250-snow-common.dtsi
@@ -446,6 +446,21 @@ trackpad@67 {
 		interrupt-parent = <&gpx1>;
 		wakeup-source;
 	};
+
+	/* Atmel mXT which is used in some revisions */
+	trackpad@4b {
+		compatible = "atmel,maxtouch";
+		reg = <0x4b>;
+		interrupt-parent = <&gpx1>;
+		interrupts = <2 IRQ_TYPE_EDGE_FALLING>;
+		wakeup-source;
+		linux,gpio-keymap = <KEY_RESERVED
+				     KEY_RESERVED
+				     KEY_RESERVED	/* GPIO0 */
+				     KEY_RESERVED	/* GPIO1 */
+				     KEY_RESERVED	/* GPIO2 */
+				     BTN_LEFT>;		/* GPIO3 */
+	};
 };
 
 /*
-- 
2.34.1

